assets/js/components/notifications/BannerNotification/index.js

```diff
@@ -212,7 +213,8 @@ export default function BannerNotification( props ) {
 		const { value: dismissed } = await getItem( cacheKeyDismissed );
 
 		if ( dismissed ) {
-			const expiration = stringToDate( dismissed );
+			const expiration = new Date( dismissed );
```

Comment:
This is a fix, as it was incorrectly changed in https://github.com/google/site-kit-wp/pull/5880. `dismissed` is a UTC date string (the result of stringifying the new `Date` object in `persistDismissal()` elsewhere in this file, see the snippet below), not a string in the format `YYYY-MM-DD` that `stringToDate()` expects. As it's a UTC date string, we should not run into timezone issues creating a new Date object from it. Besides which, the value was set from the current browser so it should be pretty safe anyway.

https://github.com/google/site-kit-wp/blob/0a2f46c56723c1fa6820c48f25ba065ad08c6197/assets/js/components/notifications/BannerNotification/index.js#L108-L109